CorPath questionnaires - Do not merge, demo only

sashaandjic opened the pull request:
To test, build this branch and run the `heracles` project.

sashaandjic committed:
CorPath questionnaires

marta- committed:
Added CorPath Logo

marta- committed:
Update AppName.json

marta- committed:
Update Media.json

marta- committed:
Updated theme colors and logo sizes

marta- commented on heracles-resources/clinical-data/src/main/media/SLING-INF/content/libs/cards/resources/media/heracles/CorPath_Light_Logo.png:
I'm not a fan of how the tagline of the logo looks. Can we ask if we can break the line after "health" and remove the dash and exclamation mark? @Hajany 

```
DISCOVER YOUR CARDIAC HEALTH
RIGHT FROM YOUR WRIST
```

**Before:**

![image](https://github.com/user-attachments/assets/35ba0f7c-d1ad-4142-af8a-c61e60a4b33a)

**After:**

![image](https://github.com/user-attachments/assets/becfb4b8-564a-42e9-872a-290373fa2bab)

Hajany replied:
I agree with the revisions, looks better this way. Let's do it and then when I demo it next week I'll confirm they're okay with it. Shouldn't be any issues with tweaking it

marta- committed:
Alternate logo with small tagline tweak

sashaandjic committed:
CorPath questionnaires - cosmetic changes to improve ui; some question name changes; spelling errors

Hajany commented:
Baseline Health Information: all the following fields are missing/conditional questions are not showing up as expected
![image](https://github.com/user-attachments/assets/9757cb50-d028-4acf-8147-f8c88343e1fd)

Hajany commented:
Baseline Health Information: conditional questions disappear once more than one option is selected
[screen-capture (9).webm](https://github.com/user-attachments/assets/ee45a00b-6787-4bb6-9a7d-c5a2ff7710f5)

Hajany commented:
Baseline Health Information: same as above, conditional questions disappear once more than one option is selected with conditional logic
[screen-capture (10).webm](https://github.com/user-attachments/assets/644a385f-205b-43b8-92c2-80266e36fcb6)

Hajany commented:
Baseline Health Information: again same as above, another use case where conditional section disappears if more than one answer option is selected
[screen-capture (11).webm](https://github.com/user-attachments/assets/58ec457f-52a3-4589-8ee7-0c9d47df3939)

Hajany commented:
2 none of the above options
![image](https://github.com/user-attachments/assets/db04c272-6de3-4ec2-ac83-d7d68d5245ac)

sashaandjic commented:
@Marta - we do not seem to support min/max for dates; should I go with text type + regexp instead?
> Min and max values are not enforced for Year of Birth question in Demographics. Min value (1943) and max value (1993) ![image](https://private-user-images.githubusercontent.com/96542282/390840656-e60cd8bf-1468-452a-abe4-0681b86619fb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI4MTIxNDYsIm5iZiI6MTczMjgxMTg0NiwicGF0aCI6Ii85NjU0MjI4Mi8zOTA4NDA2NTYtZTYwY2Q4YmYtMTQ2OC00NTJhLWFiZTQtMDY4MWI4NjYxOWZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI4VDE2MzcyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3YjliNzBjZDgwM2Q2NTAzNjc5Y2I1ZWQ2NzgzNmQ0N2RjOGQ0MmFhZDZlYWI4NjhkYjU5MjZjYjFiZTkxMTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FItFmOzrirkalznRF_1q0TTnqgJbRV-z5Is35krPzpA)

Hajany commented:
Baseline Medications: remove one Diabetes Medication answer option as it's repeated twice, keep format of "e.g." consistent replace "eg" with "e.g." and replaced highlighted "-" with brackets "( )"
![image](https://github.com/user-attachments/assets/cdc0e6a0-a2e0-4401-9175-62fab008b085)

Hajany commented:
Baseline Medications: this question should not appear if either beta blocker or calcium channel blocker is selected in previous question. Currently, it does appear
![image](https://github.com/user-attachments/assets/dc751834-3c93-4864-ad5b-03b6146e5def)
![image](https://github.com/user-attachments/assets/e13590e3-013c-4c2a-ae27-ed636a78035d)

Hajany commented:
Baseline Medications: all of these conditional questions are missing/not showing up if Beta Blocker is selected
![image](https://github.com/user-attachments/assets/fc054cb4-b3d1-4716-9d0b-18f1749132db)

Similarly, all of these conditional questions are missing/not showing up if Channel Blocker is selected
![image](https://github.com/user-attachments/assets/0273434d-3b3f-42eb-bb3c-38536323382f)

sashaandjic committed:
Baseline Health Iformation: replaced comarator "=" with "includes" for all multivalued conditionals; removed an extra None of the above option from Family Hx question

Hajany commented:
Historical Lab Results: missing units 

- white blood cells description should be **4,500 (4.5) to 11,000 (11.0)** 
- red blood cells description should be Male range: 4.7 - 6.1 **million**, Female range: 4.2 - 5.4 **million**
- hemoglobin description should be Male range: 140 - 180 **g/L**, Female range: 120 - 160 **g/L**

![image](https://github.com/user-attachments/assets/c82c34b0-870b-4b16-9e45-b8ef95648160)

Hajany commented:
Historical Lab Results: range is not working as expected, range for this question is 12-40, anything outside this range should produce a warning 
![image](https://github.com/user-attachments/assets/3d2675a0-d6a4-4a00-bc6b-37502ea1a1e5)

Hajany commented:
Historical Lab Results: description edits

- Albumin, Urine: severely increase is **300mg/g or higher** (right now it reads only as greater than 300)
- Creatinine, Urine: edit to **Normal for adult male** and **Normal for adult female**
- Creatinine-Albumin, Urine: edit to **Normal for adult male** and **Normal for adult female** and male should be **<17 mg/g** and female should be **<25 mg/g**

![image](https://github.com/user-attachments/assets/486a1500-8ed3-44da-903e-060c0aeec12a)

sashaandjic committed:
Baseline Medications: remove one Diabetes Medication answer option as it's repeated twice, keep format of "e.g." consistent replace "eg" with "e.g." and replaced highlighted "-" with brackets "( )"

sashaandjic commented:
> Baseline Medications: remove one Diabetes Medication answer option as it's repeated twice, keep format of "e.g." consistent replace "eg" with "e.g." and replaced highlighted "-" with brackets "( )"

addressed in https://github.com/data-team-uhn/cards/pull/1849/commits/d73489f490cf1e7d5ee7fc33a48a316f180f084a

Hajany commented:
Participant Status: the Point of Entry question should be conditional on if Cohort 2 is selected above
![image](https://github.com/user-attachments/assets/d75a289c-b296-4f7c-ac70-bf8e3070ad5c)
![image](https://github.com/user-attachments/assets/536e882d-2345-4929-821b-c082274f0395)

Hajany commented:
Re versions of CCTA form, I like CCTA2!

sashaandjic commented:
> Re versions of CCTA form, I like CCTA2!

CCTA2 is implemented as per original requirement.

sashaandjic committed:
Participant status: added condition for Point of entry; moved Cohort and Point of entry up (below Participant status)

sashaandjic commented:
> Participant Status: the Point of Entry question should be conditional on if Cohort 2 is selected above

fixed in https://github.com/data-team-uhn/cards/pull/1849/commits/f9019cbba3384ed227d97411bb8d470aa20be24a

![image](https://github.com/user-attachments/assets/51b9a98f-6ac8-419b-9670-27267aad5ab5)

sashaandjic committed:
Albumin, Urine: severely increase is 300mg/g or higher (right now it reads only as greater than 300)
Creatinine, Urine: edit to Normal for adult male and Normal for adult female
Creatinine-Albumin, Urine: edit to Normal for adult male and Normal for adult female and male should be <17 mg/g and female should be <25 mg/g

marta- commented:
> @marta - we do not seem to support min/max for dates; should I go with text type + regexp instead?
> 
> > Min and max values are not enforced for Year of Birth question in Demographics. Min value (1943) and max value (1993) 

Or a number (long) field with minValue and maxValue

sashaandjic commented:
> Historical Lab Results AND in Lab Results: description edits
> 
> * Albumin, Urine: severely increase is **300mg/g or higher** (right now it reads only as greater than 300)
> * Creatinine, Urine: edit to **Normal for adult male** and **Normal for adult female**
> * Creatinine-Albumin, Urine: edit to **Normal for adult male** and **Normal for adult female** and male should be **<17 mg/g** and female should be **<25 mg/g**

fixed in https://github.com/data-team-uhn/cards/pull/1849/commits/8006ecc7bc141f549060db510296b94b3a378102

![image](https://github.com/user-attachments/assets/ea529e2c-4f00-4e39-a804-14666ec44378)

sashaandjic committed:
Historical Lab Results AND in Lab Results: missing units

white blood cells description should be 4,500 (4.5) to 11,000 (11.0)
red blood cells description should be Male range: 4.7 - 6.1 million, Female range: 4.2 - 5.4 million
hemoglobin description should be Male range: 140 - 180 g/L, Female range: 120 - 160 g/L

sashaandjic commented:
> Historical Lab Results AND in Lab Results: missing units
> 
> * white blood cells description should be **4,500 (4.5) to 11,000 (11.0)**
> * red blood cells description should be Male range: 4.7 - 6.1 **million**, Female range: 4.2 - 5.4 **million**
> * hemoglobin description should be Male range: 140 - 180 **g/L**, Female range: 120 - 160 **g/L**

fixed in https://github.com/data-team-uhn/cards/pull/1849/commits/e218a343bc75e0bb9100845bc589b6794835d41b

![image](https://github.com/user-attachments/assets/d4a0fd12-4aa1-4068-9099-ebaba65d2a66)

sashaandjic commented:
> Historical Lab Results AND in Lab Results: range is not working as expected, range for this question is 12-40, anything outside this range should produce a warning ![image](https://private-user-images.githubusercontent.com/96542282/390888903-3d2675a0-d6a4-4a00-bc6b-37502ea1a1e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI4NDgyNjIsIm5iZiI6MTczMjg0Nzk2MiwicGF0aCI6Ii85NjU0MjI4Mi8zOTA4ODg5MDMtM2QyNjc1YTAtZDZhNC00YTAwLWJjNmItMzc1MDJlYTFhMWU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI5VDAyMzkyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkMTQ4MWU4MmFjZjg5NGVlYmNlMzA1NjgwMmQ0NTliMjZjODJlYWY5ZjQ2ZDM4NDU4ODBkOThlYTJmMzJjYmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SqwPlrpBANDJVjaLTN8_nfc1bl9Fun16UkoLisia2zE)

The warning is there

sashaandjic committed:
Historical Lab Results AND in Lab Results: range is not working as expected, range for this question is 12-40, anything outside this range should produce a warning

sashaandjic commented:
> Historical Lab Results AND in Lab Results: range is not working as expected, range for this question is 12-40, anything outside this range should produce a warning

fixed in https://github.com/data-team-uhn/cards/pull/1849/commits/ef3dffff48ee4777a12c5c8e6f79ba171c552a58

sashaandjic committed:
Lab Results and Historical Lab results: corrected mistakes made in the previous commit

sashaandjic commented:
![image](https://github.com/user-attachments/assets/d7158b24-a845-47bf-8fdf-bef035bdb528)

sashaandjic committed:
Min and max values are not enforced for Year of Birth question in Demographics. Min value (1943) and max value (1993)

sashaandjic commented:
> Min and max values are not enforced for Year of Birth question in Demographics. Min value (1943) and max value (1993)

fixed in https://github.com/data-team-uhn/cards/pull/1849/commits/7fe6c242ee75769ec5120218f8b14f8b7fc411cb

![image](https://github.com/user-attachments/assets/d9064792-1df1-40f7-ab1b-829660443ba4)

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Demographics.xml:
```diff
@@ -40,24 +45,24 @@
 			<name>description</name>
 			<value>Format: YYYY</value>
 			<type>String</type>
 		</property>
```
I would remove this description. The area under this question is getting busy with all the answer instructions, and we already know 4 digits are expected because we request values between 1943 and 1993.

sashaandjic replied:
Done. 
![image](https://github.com/user-attachments/assets/450189bd-b309-4e21-ace7-8a3b730bfc45)

sashaandjic committed:
Demographic form: removed "Format: YYYY" under Year of birth

sashaandjic committed:
CCTA - Location question: Added descripiton under Uspecified and Not Reported options, instead of showing them under Location title

sashaandjic committed:
CCTA2 - Location qustions: Added decription under Unspecified and Not Reported options, instead of showing them under the Location title.

Hajany commented:
Baseline Health Information: **Date of CVD diagnosis** is not conditional on anything, currently it only appears if "Other CV diagnoses" is selected but it should appear if **any** relevant diagnoses is selected

![image](https://github.com/user-attachments/assets/c67ee99a-302c-46de-b518-0c6b8556c7fe)
![image](https://github.com/user-attachments/assets/07de9d68-0664-448e-844b-b9effdaa2f5b)

sashaandjic committed:
Baseline Health Information: removed conditional from Date od CVD diagnosis (bl_cvd_date)

sashaandjic commented:
[Fix ](https://github.com/data-team-uhn/cards/pull/1849/commits/43426ebedfd1c5e44150f0fbf4f5f25ecf54bd89)for: Baseline Health Information: Date of CVD diagnosis is not conditional on anything, currently it only appears if "Other CV diagnoses" is selected but it should appear if any relevant diagnoses is selected

![image](https://github.com/user-attachments/assets/dae3ef90-9cf3-4364-b27e-224d0d226288)

Hajany commented:
> > Historical Lab Results AND in Lab Results: description edits
> > 
> > * Albumin, Urine: severely increase is **300mg/g or higher** (right now it reads only as greater than 300)
> > * Creatinine, Urine: edit to **Normal for adult male** and **Normal for adult female**
> > * Creatinine-Albumin, Urine: edit to **Normal for adult male** and **Normal for adult female** and male should be **<17 mg/g** and female should be **<25 mg/g**
> 
> fixed in [8006ecc](https://github.com/data-team-uhn/cards/commit/8006ecc7bc141f549060db510296b94b3a378102)
> 
> ![image](https://private-user-images.githubusercontent.com/23243244/390942571-ea529e2c-4f00-4e39-a804-14666ec44378.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyNjIzNjUsIm5iZiI6MTczMzI2MjA2NSwicGF0aCI6Ii8yMzI0MzI0NC8zOTA5NDI1NzEtZWE1MjllMmMtNGYwMC00ZTM5LWE4MDQtMTQ2NjZlYzQ0Mzc4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAzVDIxNDEwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0YTA1ODkyZTRkZTczYzc3OGIzM2IzMzg0NGY3NWZiMzQ3NDBkOWNmZGExODQ1ODUzZjQwNDg2NzQxYmU0YTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JTS8o0XYUe2ot8RZXAvnkbGD_W560E22YDmLxc64UaY)

Creatinine-Albumin, Urine: edit to female should be **<25 mg/g** (missing less than sign)

sashaandjic committed:
Creatinine-Albumin, Urine: edit to female should be <25 mg/g (was missing less than sign)

sashaandjic commented:
> Creatinine-Albumin, Urine: edit to female should be **<25 mg/g** (missing less than sign)
Fixed in this [commit](https://github.com/data-team-uhn/cards/pull/1849/commits/ad29f0affb76302900c943348298456574e9dfd9)
![image](https://github.com/user-attachments/assets/64d7791a-8156-4fc6-81d6-d8006314e2fc)

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/CCTA2.xml:
```diff
@@ -0,0 +1,8258 @@
+			<value>Filename: CP######_ccta_report.pdf</value>
+			<type>String</type>
+		</property>
+		<property>
+			<name>minValueLabel</name>
+			<value>Must be pdf file extension
+ID must match database profile ID
+Filename must match: CP######_ccta_report.pdf</value>
+			<type>String</type>
+		</property>
```
This property was most likely entered in error and should be removed. Should the text in the `value` be moved up to the `description` property or is the description fine as is?

Same issue in CCTA.xml (unsure which one we'll end up using).

sashaandjic replied:
Oh, I mistakenly left that text in the spreadsheet in the minValueLabel column. I will make a fix for both: CCTA  and CCTA2.

sashaandjic committed:
Removed strange minValueLabel property from CCTA and CCTA2 forms

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
CCTA2 and Baseline Medications changes

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Demographics and Physical Assessments

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Baseline Health info and PArticipant Status forms: max date today; pack of ciggarettes max value 80; insufficient.. -> Physically inactive lable change

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Lab REsults and Historical Lab results: for date allow today only; removed pro-BNP question

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Baseline Medications.xml:
```diff
@@ -12400,4 +12544,482 @@ If only year is known, code the month and day as 01.</value>
 		</node>
 	</node>
+	<node>
+		<name>medications_Phosphodiesterasenumber5inhi</name>
```
they'll probably want the full word: `medications_Phosphodiesterasenumber5inhibitors`

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Baseline Medications.xml:
```diff
@@ -12400,4 +12544,482 @@ If only year is known, code the month and day as 01.</value>
+			</node>
+		</node>
+		<node>
+			<name>medications_Phosphodiesterasenumber5inhi_bl_med_number</name>
```
Here as well: `medications_Phosphodiesterasenumber5inhibitors_bl_med_number`

Also in all related quesions added in this commit.

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/CCTA2.xml:
```diff
@@ -75,13 +75,18 @@
 		<property>
 			<name>label</name>
-			<value>Clinical Indication</value>
+			<value>Report Section Header:CLINICAL INDICATION</value>
```
Missing space after ":"

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Small changes that were missed or were reported by reviewers

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Demographic: made Ethnicity multiselect; properly added None of the above and Prefer not to answer options

marta- reviewed (approved):
looks good to me, didn't test

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Lab results and Hist Lab result edits

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Fixes in Historical Lab results: all minAnswers should be 0, except for the lab_date (the only required question)

Hajany commented:
CAD is still present in list of options, remove Coronary artery disease (CAD) option
![image](https://github.com/user-attachments/assets/823074ec-b313-4fe9-8bab-bc978c0c9316)

Hajany commented:
Should be "specify other cardiac cath therapy"
![image](https://github.com/user-attachments/assets/f8facd07-f1c1-42d6-9691-3ac028cd3ad3)

Hajany commented:
Historical Lab Results + Lab Results: wrong units 

- Creatinine: unit should be mmol/L
- Albumin:Creatinine Ratio: unit should be mg/mmol Cr

![image](https://github.com/user-attachments/assets/30889be4-6b5a-491c-b556-dc32b9d3b479)
![image](https://github.com/user-attachments/assets/22d60804-8087-4fed-ab04-ab9d1c58b14a)

Hajany commented:
Historical Lab Results + Lab Results: can we make the "2" a superscript? 
![image](https://github.com/user-attachments/assets/e8c8c6ac-3f70-4a0c-87e4-ddbfcb06c2e3)
![image](https://github.com/user-attachments/assets/a8e8c40c-26ab-4b84-8617-ee9ec5b1c6e4)

Hajany commented:
CCTA2: edit **Dominance** and **Other** to have a lowercase d and o
![image](https://github.com/user-attachments/assets/a83000cb-cb3f-495b-916c-2d8f01ba473f)

Hajany commented:
Since we agreed to proceed with the CCTA2 implementation, can we remove CCTA from the list and rename CCTA2 to CCTA?

marta- commented:
> * Albumin:Creatinine Ratio: unit should be mg/mmol Cr

@Hajany should the units also be changed in the description?

marta- commented:
> Historical Lab Results + Lab Results: can we make the "2" a superscript?

We don't support that for units yet, but we absolutely should. I'll make a task.

marta- commented:
> CCTA2: edit **Dominance** and **Other** to have a lowercase d and o

There's so much random title-casing allover the forms, including in descriptions and answer options. Should we go over everything make it all sentence case?

marta- committed:
CORPATH-5: Implement feedback from Phase 1 forms demo

Addressing comments:
* BHI - removed CAD option, fixed question text
* CCTA - fixed letter casing
* HLR - updated some units of measurement

marta- committed:
CORPATH-5: Implement feedback from Phase 1 forms demo

Addressing comments: removed unused CCTA.xml, changed the node name and title of CCTA2 to CCTA. The file CCTA2.xml will be renamed to CCTA.xml in the new corpath repo.

marta- commented:
> Since we agreed to proceed with the CCTA2 implementation, can we remove CCTA from the list and rename CCTA2 to CCTA?

0cc4d67 : Removed CCTA.xml and renamed the _node_ in CCTA2.xml, but not the file itself (yet). I suggest we rename the file when we're done with the tweaks and move the questionnaire to the new repo. When a file is renamed, it's history is not as easily accessible, and I find it useful to see the changes for now.

Hajany commented:
> > CCTA2: edit **Dominance** and **Other** to have a lowercase d and o
> 
> There's so much random title-casing allover the forms, including in descriptions and answer options. Should we go over everything make it all sentence case?

Yes, let's make everything sentence case (except for the section headings, in which they had a specific format)

marta- committed:
CORPATH-5: Implement feedback from Phase 1 forms demo

Addressing comments: made 2 superscript in unit of measurement

marta- committed:
CARDS-2613: Units of measurement for numeric answers should support superscript (e.g. m^2)

marta- commented:
> > Historical Lab Results + Lab Results: can we make the "2" a superscript?
> 
> We don't support that for units yet, but we absolutely should. I'll make a task.

Functionality implemented in #1861 (also available on this branch). Questionnaire change made in bb4775c .

Hajany commented:
> > * Albumin:Creatinine Ratio: unit should be mg/mmol Cr
> 
> @Hajany should the values in these units be added in the description?

Good catch - yes, description should be updated to the below
![image](https://github.com/user-attachments/assets/bb8c10bf-1a7c-4813-a041-1ae89c5e5d76)

- Microalbumin: Normal: < 3mg/mmol | Microalbuminuria: 3-30 mg/mmol | Macroalbuminuria: > 30 mg/mmol
- Creatinine: Normal: Male: 3.54-24.6mmol/L, Female: 2.55-20.0 mmol/L | Anything greater than these values could suggest kidney disease
- Albumin:Creatinine Ratio: Normal:  Male: <2.5mg/mmol, Female: <3.5mg/mmol | Microalbuminuria:  Male: 2.5-25mg/mmol, Female: 3.5-35mg/mmol | Macroalbuminuria: Male: 25-299mg/mmol, Female: 35-299mg/mmol | Nephrotic Syndrome: >300mg/mmol

**( | indicates a line break)

sdumitriu commented:
Should this be moved over to the new repo?

marta- committed:
CORPATH-5: Implement feedback from Phase 1 forms demo

Addressing comments: updated albumin/creatinine descriptions in HLR

marta- commented:
> > > * Albumin:Creatinine Ratio: unit should be mg/mmol Cr
> > 
> > 
> > @Hajany should the values in these units be added in the description?
> 
> Good catch - yes, description should be updated to the below 

`Normal: Male: ` (one `:` after another on the same line) looks really wrong to me. I removed the ":" after "Male" and "Female" and I find that it's still clear what it means while beeing easier on the eyes. Please let me know if they should be added back @Hajany .

![image](https://github.com/user-attachments/assets/eaa06e60-b03a-4722-91ac-8ff8d3ea3799)

Hajany commented:
> > > > * Albumin:Creatinine Ratio: unit should be mg/mmol Cr
> > > 
> > > 
> > > @Hajany should the values in these units be added in the description?
> > 
> > 
> > Good catch - yes, description should be updated to the below
> 
> `Normal: Male: ` (one `:` after another on the same line) looks really wrong to me. I removed the ":" after "Male" and "Female" and I find that it's still clear what it means while beeing easier on the eyes. Please let me know if they should be added back @Hajany .
> 
> ![image](https://private-user-images.githubusercontent.com/651980/394862888-eaa06e60-b03a-4722-91ac-8ff8d3ea3799.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5NDA3MDIsIm5iZiI6MTczMzk0MDQwMiwicGF0aCI6Ii82NTE5ODAvMzk0ODYyODg4LWVhYTA2ZTYwLWIwM2EtNDcyMi05MWFjLThmZjhkM2VhMzc5OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxMVQxODA2NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNDAwMGE1M2Q2MzRhYWU3YmQyMDE3ZDljNmI2ZDM3NmQ5MmU4ZTNiZjI2MWQ2M2IwMjFjMmUzYTNkY2RmNTJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.kjiy2JBunkPMbPNx9DuRofpTYH-hs8UkWiwz65nLVJs)

looks good

marta- commented:
@Hajany can I assume the same update to descriptions needs to be done in Laboratory results?

![image](https://github.com/user-attachments/assets/523f9900-6e9f-431d-9bd1-692b3606074a)

Hajany commented:
> Also, LR and HLR have visit numbers in their descriptions, but there are no more Visit Number fields. Should the descriptions be removed?
> 
> ![image](https://private-user-images.githubusercontent.com/651980/394866511-55b85ef9-b610-4063-a487-31283cb0826e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5NDExNTgsIm5iZiI6MTczMzk0MDg1OCwicGF0aCI6Ii82NTE5ODAvMzk0ODY2NTExLTU1Yjg1ZWY5LWI2MTAtNDA2My1hNDg3LTMxMjgzY2IwODI2ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxMVQxODE0MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NjdmYzEyM2YwNDBlNzJlNzRmNDIxMTIwNDBiYWQwNTViN2Y5ZmRiODY3MmZkYzgyNmJkNGU1YjllODRjZTMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.KMFIiC_zHskcIcF7c-Y1lEN0z93nfnUVUgjqhSSLGPg)

yes remove visit numbers in the description

marta- committed:
CORPATH-5: Implement feedback from Phase 1 forms demo

Updated units of measurement and limit descriptions in LR

marta- committed:
CORPATH-5: Implement feedback from Phase 1 forms demo

Remove mention of visit numbers from the descriptions of LR and HLR

marta- committed:
CORPATH-5: Implement feedback from Phase 1 forms demo

Typo fix

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Turned Title Case into Sentence case for option Not Reported -> Not reported

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Turned Title Case into Sentence case for all question texts, descriptions and option labels

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Final minor edits; added max number of forms per subject

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Lab Results form: added proper units for Estimated glomerular filtration rate

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Baseline medications: remove dash from Patient-reported in the question description

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Baseline Healt Information: removed form's description property

marta- commented:
> Remove space before and after the / 


Actually:

> Space before and after slash
> Many people wonder whether to use a space before and after slashes. It depends. When a slash signifies alternatives between only two words, don’t use spaces before or after. 

> 
> When using slashes to signify alternatives between phrases or multi-word terms or [compounds](https://www.grammarly.com/blog/parts-of-speech/compound-adjectives/), a space before and after the slash makes text easier to read. Another time it’s acceptable to use a space after a slash is when breaking up lines of a poem, song, or play. In this case, a line break should be made after the slash.

> 
> Example: Add chili flakes and/or black pepper to the recipe.

> 
> Example: World War I / First World War

marta- commented:
I would also revert all Ethnicity options to spaced slashes @sashaandjic @Hajany

Hajany commented:
Historical & Lab Results: spelling error, should read as Microalbuminuria and Macroalbuminuria
![image](https://github.com/user-attachments/assets/fc6405eb-7dd9-432f-a5fc-4a8c6dd4c7d6)

sashaandjic committed:
CORPATH-5: Implement feedback from Phase 1 forms demo
Spelling errors in Historical Lab and Laboratory; added spaces in Demographic ethnicities before and aflter slash

Hajany reviewed (approved):
looks good

marta- committed:
CORPATH-7: Legacy Data form - update to format of PL#######

marta- commented:
Closing because we moved the questionnaires to the `corpath` repo: https://github.com/data-team-uhn/corpath/pull/1 .
